### ISSUE TITLE
Pause not working

### ISSUE THREAD

**Issue by ajorgensen** [Burgestrand/hallon-openal#3]
Hi

When I load a track and begin playing it with

```
track.load
player.play(track)
```

and then try to pause the track using

```
player.pause
```

the track will audibly pause but then when I try to resume playback the song is significantly farther along in the song than when it was first paused. The position in the song seems to be directly related to the time between the time it was paused and when it was resumed. Am i missing something?

**Comment by Burgestrand** [comment]
Hm, I can’t reproduce this I’m afraid.

When playing sound with Hallon, there are two things that are taken into account on how much audio you’ve played. On one hand, you’re streaming audio (how much audio you have available to play), and on another you’re playing audio (how much audio you have consumed).

Hallon itself will buffer about one second of audio, and Hallon::OpenAL can buffer about three seconds. So, worst case you could only have advanced about four seconds into the song, assuming libspotify did stop streaming when asked.

So, assuming you pause the song and play it back 10-15 seconds later, have it advanced for all that time, or just 2-4 seconds?

**Comment by ajorgensen** [comment]
It seems to advance for the whole 10-15 seconds. So in that case by your explanation of how things work it sounds like libspotify isn't actually stopping the stream.

**Comment by Burgestrand** [comment]
Alright! So, assuming libspotify does not stop streaming when asked, when you later continue playing the song it will play where it was for about a second (possibly slightly more, but no more than four seconds), and then suddenly skip forward all the time that you had paused. Does this sound like an accurate description of what’s happening for you?

I’ve tried reproducing this at different workstations at work but have been unsuccessful I’m afraid. Could you disclose some details about your operating system, ruby version, version of Hallon and version of libspotify? I’m not sure either of those would help, but if I’m to track this down it ought to help if I do it under a similar environment.

If you bear with me, there are two things I’d like to ask you to try:
- remove the `tmp/` directory created in your current working directory when creating the Hallon session. This directory contains the libspotify cache, and libspotify is known to behave _very_ strangely when this cache is damaged in any way.
- if you have the possibility to, do run the code on another computer and see if the error can be reproduced there as well. It might help narrow the issue down further for me.

By the way, thanks for reporting this issue. I’m sincerely grateful. :)

**Comment by Burgestrand** [comment]
Thought about it some more. libspotify should not continue into the song after pausing anyway, since Hallon's buffer would fill up and not accept any more audio. libspotify will know that the buffer is full on music delivery, and should resend the last piece of audio every time after that until Hallon reports to libspotify that the audio was accepted.

So, if libspotify would skip this much audio the bug in libspotify (or Hallon) would be bigger than I initially thought.

TL;DR: weird!

**Comment by Burgestrand** [comment]
@ajorgensen How is this issue coming along for you?
